File: arrayprocessing.rb
#arrayprocessing.rb prints the values of an array and reads in the values of an array
# Lorrayya Williams
# March 11, 2019 for CS214
#################################################################



# Input: An array
# The array has contents
# Output: The values of the array
################################################################
def printArray(anArray)
  x = 0
  puts "The array values will be printed below"
  while x < anArray.length
    puts "#{anArray[x]} \n"
    x += 1
  end
end


# Input: An array
# Precondition: The user inputs numbers
# Output a modified array
###############################################################
def readArray(anArray)
  x = 0
  while x < anArray.length
    puts "Add a value to the Array."
    anArray[x] = gets.to_f
    x += 1
  end
end



def main
print "Hello! This program allows you to create, add values to, and print an array \n";
print "What size do you want your array? ";
arraysize = gets.to_i
array = Array.new(arraysize)
readArray(array)
printArray(array)
end

main

File: arrayprocessing.java
/* Arrayprocessing.java is a driver for methods printArray and readArray.
 * Student Name: Lorrayya Williams
 * Date: March 08, 2019 for CS 214 at Calvin College
 * Precondition: theArray, is an array of double values.
 * Output: the average of the numbers in theArray.
 ***************************************************************/
 import java.io.*;
 import java.util.Scanner;

public class arrayprocessing
{
  public static void main(String[] args)
    {
  Scanner keyboard = new Scanner(System.in);
  System.out.println("Hello! This programs allows you to create, add values to, and print an array.");
  System.out.println("How many values would you like to be in the array?");
  int value = keyboard.nextInt(); //inputs the size of array
  Double [] values = new Double[value]; //creates the array

  System.out.println("What values would you like in the array?");
    readArray(values);
    printArray(values);

}


/* Input: Accepts an Array
* Precondition: The array contains values.
* Output: Prints the values of the Arrayprocessing
******************************************************/
  public static void printArray(Double [] anArray){
    System.out.println("The values of the array are going to be printed below.");
    for(int i=0; i < anArray.length; i++){
      System.out.println(anArray[i] + "\n");
    }
  }


/* Input: An Array and the values within the array.
* Precondition: The inputed values are numerical
* Output: A modified array.
********************************************************/
  public static void readArray(Double [] anArray){
      Scanner keyboard = new Scanner(System.in);
      for(int i =0; i < anArray.length; i++){
      System.out.println("Please add a value to the array: ");
      anArray[i] = keyboard.nextDouble();
      }
  }
}
